src/main/java/edu/upenn/library/solrplugins/JsonReferencePayloadHandler.java

```diff
@@ -12,33 +11,81 @@
 import org.apache.solr.request.FacetPayload;
 
 /**
- * Builds facet payloads for fields tokenized by JsonReferencePayloadTokenizer.
+ * Builds facet payloads from fields containing normalized, filing, and
+ * prefix strings joined by a delimiter, and a payload attribute
+ * containing information about references.
  *
  * The NamedList structure for a facet will look like this:
  *
  * <lst name="subject_xfacet">
  *   <lst name="Hegelianism">
  *     <int name="count">3</int>
- *     <lst name="see_also">
- *       <long name="Georg Wilhelm Friedrich Hegel">1</long>
- *       <long name="History of Marxism">1</long>
- *     </lst>
- *   </lst>
- *   <lst name="Georg Wilhelm Friedrich Hegel">
- *     <int name="count">2</int>
- *     <lst name="see_also">
- *       <long name="History of Marxism">1</long>
+ *     <str name="normalized">hegelianism</str>
+ *     <str name="prefix"></str>
+ *     <str name="filing">Hegelianism</str>
+ *     <lst name="refs">
+ *       <lst name="see_also">
+ *         <lst name="Georg Wilhelm Friedrich Hegel">
+ *           <long name="count">1</int>
+ *           <str name="normalized">hegel</str>
+ *           <str name="prefix">Georg Wilhelm Friedrich </str>
+ *           <str name="filing">Hegel</str>
+ *         </lst>
+ *       </lst>
  *     </lst>
  *   </lst>
  * </lst>
  *
  * @author jeffchiu
  */
 public class JsonReferencePayloadHandler implements FacetPayload<NamedList<Object>> {
+  private static final String DELIM = "\u0000";
+  private static final String KEY_REFS = "refs";
+  private static final String KEY_NORMALIZED = "normalized";
+  private static final String KEY_PREFIX = "prefix";
+  private static final String KEY_FILING = "filing";
+
+  /**
+   * overwrite entry in NamedList with new value
+   * (update existing key, or add the key/value if key doesn't already exist)
+   */
+  private static void overwriteInNamedList(NamedList<Object> namedList, String key, Object value) {
+    int indexOfKey = namedList.indexOf(key, 0);
+    if(indexOfKey != -1) {
+      namedList.setVal(indexOfKey, value);
+    } else {
+      namedList.add(key, value);
+    }
+  }
+
+  /**
+   * Copies a field value from one NamedList into another
+   */
+  private static void copyField(NamedList<Object> from, NamedList<Object> to, String key) {
```

Comment:
"copyField" is [overloaded in a Solr context](https://cwiki.apache.org/confluence/display/solr/Copying+Fields) ... consider changing this method name to avoid confusion?